pkg/abstractions/image/build.go

```diff
@@ -199,6 +199,9 @@ func (b *Builder) Build(ctx context.Context, opts *BuildOpts, outputChan chan co
 		return err
 	}
 
+	// Get the hostname of the build container
+	go b.checkForInitialImagePullError(containerId, outputChan)
```

Comment:
I think we can probably do this without an extra goroutine

Reply:
Ideally I check in the gateway to see if the image exists or not. But the packages that help do this require a shitton of other packages